app/components/provider_interface/course_summary_component.rb

```diff
@@ -30,6 +30,17 @@ def rows
           value: study_mode,
         },
       ]
+      return rows unless course_option.course&.accredited_provider&.name_and_code
+
+      rows << accredited_body_details(course_option)
+    end
+
+  private
+
+    def accredited_body_details(course_option)
+      { key: 'Accredited body',
+        value: course_option.course&.accredited_provider&.name_and_code,
```

Comment:
Now this is its own method, the accredited_provider presence check is the responsibility of the caller.

```suggestion
        value: course_option.course.accredited_provider.name_and_code,
```